Mac-App/Backend/Backend/Shared/Extensions/DispatchQueue.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  DispatchQueue.swift
+//  Backend
+//
+//  Created by Daniel Garcia on 28.10.20.
+//  Copyright © 2020 Acphut Werkstatt. All rights reserved.
+//
+
+extension DispatchQueue {
+    func asyncWithCheck(
+        key: DispatchSpecificKey<String>,
+        execute work: @escaping () -> Swift.Void
+    ) {
+        if let _ = DispatchQueue.getSpecific(key: key) {
+            work()
+        } else {
+            async {
+                work()
+            }
+        }
+    }
+}
```

Comment:
Can we rename the method so it is obvious what are we really checking? Or a comment perhaps?

Reply:
Yeah, I was struggled naming this. Do you have any suggestions?

Reply:
What do you think of this? 
https://github.com/Thurman1776-/Uncomplect-us/pull/28/commits/14d827f302f8514b3cc70468566d5b4d8031b52a

Reply:
Nice one @Thurman1776-